### PR TITLE
AWS, Core: Slim down Jetty config for tests

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java [hunk 1]
```diff
@@ -171,12 +171,8 @@ private static Server initHttpServer() throws Exception {
...
     ServletContextHandler servletContext =
         new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
-    servletContext.setContextPath("/");
-    ServletHolder servletHolder = new ServletHolder(servlet);
-    servletHolder.setInitParameter("javax.ws.rs.Application", "ServiceListPublic");
-    servletContext.addServlet(servletHolder, "/*");
-    servletContext.setVirtualHosts(null);
-    servletContext.setGzipHandler(new GzipHandler());
+    servletContext.addServlet(new ServletHolder(servlet), "/*");
+    servletContext.setHandler(new GzipHandler());
```
Comment (nastra) [inline aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java:175]: `setGzipHandler` is deprecated, so switching to `setHandler` here and in other places

File: aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java [hunk 2]
```diff
@@ -171,12 +171,8 @@ private static Server initHttpServer() throws Exception {
         new S3SignerServlet(S3ObjectMapper.mapper(), ImmutableList.of(deleteObjectsWithBody));
     ServletContextHandler servletContext =
         new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
-    servletContext.setContextPath("/");
```
Comment (nastra) [inline aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java:174]: those things don't seem to be required for passing tests, so we can slim down the configuration a bit

**Review by Fokko — APPROVED** [review]
Less is more!

File: aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java (hunk 1 above)
Comment (amogh-jahagirdar) [reply aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java:175]: Yeah I realized setting the Gzip handler is important, I thought it was already done by default but apparently not!

**Review by amogh-jahagirdar — APPROVED** [review]
LGTM, thanks @nastra !

**Pull request merged**